### PR TITLE
Add orbit_ggp::Project struct

### PR DESCRIPTION
**Pull request by antonrohr**
http://b/197603266

### CODE REVIEW

File: src/OrbitGgp/ProjectTest.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  {
+    // one partial (invalid) element
+    // pretty json:
+    // [
+    //  {
+    //   "displayName": "display name"
+    //  }
+    // ]
+    const auto json = QString(R"([{"displayName":"display name"}])").toUtf8();
```
Comment (karupayun) [inline src/OrbitGgp/ProjectTest.cpp:44]: Isn't cleaner to have a variable name with that json and try to parse it directly? 

(Something like that but I'm not sure how to write a multi-line string)
```
const Qstring kOnePartialInvalidElement = (R"(
     [
      {
       "displayName": "display name"
      }
    ])"
const auto json = kOnePartialInvalidElement.toUtf8();
```

**Review by vickyliu-go4it — APPROVED** [review]
LGTM!

File: src/OrbitGgp/ProjectTest.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const auto result = Project::GetListFromJson(json);
+    ASSERT_THAT(result, orbit_test_utils::HasNoError());
+    const QVector<Project>& projects{result.value()};
+    ASSERT_EQ(projects.size(), 2);
+    EXPECT_EQ(projects[0].display_name, "a display name");
+    EXPECT_EQ(projects[0].id, "project id");
+    EXPECT_EQ(projects[1].display_name, "second display name");
+    EXPECT_EQ(projects[1].id, "project id 2");
+  }
+}
```
Comment (karupayun) [inline src/OrbitGgp/ProjectTest.cpp:74]: Nit: Add a test where one project works and other not.

File: src/OrbitGgp/ProjectTest.cpp (hunk 1 above)
Comment (antonrohr) [reply src/OrbitGgp/ProjectTest.cpp:44]: I changed it to not have the pretty version in the comment, but the json string being pretty json on its own. It messes a bit with the formatting but its better than before. Thanks for the suggestion!

File: src/OrbitGgp/ProjectTest.cpp (hunk 2 above)
Comment (antonrohr) [reply src/OrbitGgp/ProjectTest.cpp:83]: done

File: src/OrbitGgp/ProjectTest.cpp (hunk 1 above)
Comment (karupayun) [reply src/OrbitGgp/ProjectTest.cpp]: So sorry for the formatting. :(

**Pull request merged**